code/datums/ruins/lavaland.dm

```diff
@@ -0,0 +1,175 @@
+/datum/map_template/ruin/lavaland/biodome
+	cost = 5
+	allow_duplicates = FALSE
+
+/datum/map_template/ruin/lavaland/biodome/beach
+	name = "Biodome Beach"
+	id = "biodome-beach"
+	description = "Seemingly plucked from a tropical destination, this beach \
+		is calm and cool, with the salty waves roaring softly in the \
+		background. Comes with a rustic wooden bar and suicidal bartender."
+	suffix = "lavaland_biodome_beach.dmm"
+
+/datum/map_template/ruin/lavaland/biodome/winter
+	name = "Biodome Winter"
+	id = "biodome-winter"
+	description = "For those getaways where you want to get back to nature, \
+		but you don't want to leave the fortified military compound where you \
+		spend your days. Includes a unique(*) laser pistol display case, \
+		and the recently introduced I.C.E(tm)."
+	suffix = "lavaland_surface_biodome_winter.dmm"
+
+/datum/map_template/ruin/lavaland/biodome/clown
+	name = "Biodome Clown Planet"
+	id = "biodome-clown"
+	description = "WELCOME TO CLOWN PLANET! HONK HONK HONK etc.!"
+	suffix = "lavaland_biodome_clown_planet.dmm"
+
+/datum/map_template/ruin/lavaland/cube
+	name = "The Wishgranter Cube"
+	id = "wishgranter-cube"
+	description = "Nothing good can come from this. Learn from their mistakes \
+		and turn around."
+	suffix = "lavaland_surface_cube.dmm"
+	cost = 10
+	allow_duplicates = FALSE
+
+/datum/map_template/ruin/lavaland/prisoners
+	name = "Prisoner Crash"
+	id = "prisoner-crash"
+	description = "This incredibly high security shuttle clearly didn't have \
+		'avoiding lavafilled hellscapes' as a design priority. As such, it \
+		has crashed, waking the prisoners from their cryostasis, and setting \
+		them loose on the wastes. If they live long enough, that is."
+	suffix = "lavaland_surface_prisoner_crash.dmm"
+	cost = 15
+	allow_duplicates = FALSE
+
+/datum/map_template/ruin/lavaland/seed_vault
+	name = "Seed Vault"
+	id = "seed-vault"
+	description = "The creators of these vaults were a highly advanced and \
+		benevolent race, and launched many into the stars, hoping to aid \
+		fledgling civilizations. However, all the inhabitants seem to do is \
+		grow drugs and guns."
+	suffix = "lavaland_surface_seed_vault.dmm"
+	cost = 10
+
+/datum/map_template/ruin/lavaland/ash_walker
+	name = "Ash Walker Nest"
+	id = "ash-walker"
+	description = "A race of unbreathing lizards live here, that run faster \
+		than a human can, worship a broken dead city, and are capable of \
+		reproducing by something involving tentacles? Probably best to \
+		stay clear."
+	suffix = "lavaland_surface_ash_walker1.dmm"
+	cost = 20
+	allow_duplicates = FALSE
+
+/datum/map_template/ruin/lavaland/free_golem
+	name = "Free Golem Ship"
+	id = "golem-ship"
+	description = "Lumbering humanoids, made out of precious metals, move \
+		inside this ship. They frequently leave to mine more minerals, \
+		which they somehow turn into more of them. Seem very intent on \
+		research and individual liberty, and also geology based naming?"
+	cost = 25
```

Comment:
I'm worried about making all the spawn ones so expensive, because some of my favourite experiences are when the various respawn roles interact with eachother